From: Felix Fietkau <nbd@nbd.name>
Date: Wed, 3 Oct 2012 00:00:00 +0200
Subject: [PATCH] mac80211: force scans in access point mode by default

Allow scans by default by making them forced scans by default.
This is achieved by removing the test for NL80211_SCAN_FLAG_AP.

---
 net/mac80211/cfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -2721,15 +2721,13 @@ static int ieee80211_scan(struct wiphy *
 		fallthrough;
 	case NL80211_IFTYPE_AP:
 		/*
-		 * If the scan has been forced (and the driver supports
-		 * forcing), don't care about being beaconing already.
+		 * Don't care about being already beaconing.
 		 * This will create problems to the attached stations (e.g. all
 		 * the  frames sent while scanning on other channel will be
 		 * lost)
 		 */
 		if (sdata->deflink.u.ap.beacon &&
-		    (!(wiphy->features & NL80211_FEATURE_AP_SCAN) ||
-		     !(req->flags & NL80211_SCAN_FLAG_AP)))
+		    !(wiphy->features & NL80211_FEATURE_AP_SCAN))
 			return -EOPNOTSUPP;
 		break;
 	case NL80211_IFTYPE_NAN:
